### PR TITLE
Changed chloromorphosis to work in more types of weather, only outside, and only at or above a certain light level.

### PR DESCRIPTION
**Pull request by root-reindeer-flotilla**
It only took me 2 weeks (well actually like 5 days total but I gave up for like a week and a half) but I finally fixed Chloromorphosis to work in more weather

Here's the rundown: 

1. It now works in more types of weather, I included most "light" and "medium" weather, as plants still grow  when it's raining out, as long as there's light. Which brings me to
2. Chloromorphosis only works when you're outside and the light is at least at level 12, this was the hardest thing to find, but I cross referenced it with SUNLIGHT_DEPENDENT, I figured the light should be bright enough to actually enable the ability, at least to the level of the sunlight dependency mutation.

Here are some potential issues:
1. Messy code, I am not a programmer.
2. Non ideal solution, like I said I'm not a programmer but I believe this is a good middle ground and the balance is ok. You have to sleep in the day time outside which is enough of a downside that doesn't make this overpowered in my opinion. Plus you still have to sleep the normal amount unless you have a mutation that affects that.

Something that would be better would be the food and thirst would be reduced with differing light levels and weather, but I would have no idea how to do that.

I built and tested this myself with Visual Studio in as many scenarios I could think of, outside in rain ✔️, inside ❌, outside at night ❌, outside in snow ✔️. If you have a concern with a scenario let me know and I'll test it out.

### CODE REVIEW

**Comment by patq911** [comment]
So this works fine but the style is off in some undefined way that fails the gorgon test.

File: src/player_hardcoded_effects.cpp [hunk 1]
```diff
@@ -1002,15 +1002,17 @@ void player::hardcoded_effects( effect &it )
...
+            || g->weather == WEATHER_CLOUDY || g->weather == WEATHER_SNOW ) ) {
             // Hunger and thirst fall before your Chloromorphic physiology!
             if( get_hunger() >= -30 ) {
                 mod_hunger( -5 );
             }
             if( get_thirst() >= -30 ) {
                 mod_thirst( -5 );
             }
-        }
+          }
```
Comment (Night-Pryanik) [inline src/player_hardcoded_effects.cpp]: This change is unneeded. The closing curly bracket should have the same spaces before it as the opening `if`. I.e. they should be on the same vertical line.

Comment (patq911) [reply src/player_hardcoded_effects.cpp]: Thanks, just sent a new commit for that.

Comment (patq911) [reply src/player_hardcoded_effects.cpp]: I don't know if you noticed but just moving that bracket made it fail even more. and I still don't know what the problem is. the "style" or something.

**Comment by Night-Pryanik** [comment]
My astyle says it should be styled like this:

```
if( calendar::once_every( 10_minutes ) && has_trait( trait_id( "CHLOROMORPH" ) ) &&
    g->is_in_sunlight( pos() ) ) {
    g->m.is_outside( pos() ) &&g->natural_light_level( posz() ) >= 12 &&
    ( g->weather == WEATHER_CLEAR || g->weather == WEATHER_SUNNY
      || g->weather == WEATHER_DRIZZLE || g->weather == WEATHER_RAINY || g->weather == WEATHER_FLURRIES
      || g->weather == WEATHER_CLOUDY || g->weather == WEATHER_SNOW ) ) {
        // Hunger and thirst fall before your Chloromorphic physiology!
    }
}
```

By the way, I'd declare some variable like `compatible_weather_types` and move all these `g->weather == WEATHER_FOO || g->weather == WEATHER_BAR` into this variable, so final formula should look like 

```
g->m.is_outside( pos() ) &&g->natural_light_level( posz() ) >= 12 && compatible_weather_types
```

**Comment by patq911** [comment]
I hope I did it right. I tested it out with my own build and it still worked the same way as far as I could tell. Unfortunately gorgon is down so I don't know if the style is correct.

I just kind of assumed weather being a certain type is a bool value, right? I mean it's either raining or not, and as long as those pipes || still work as "or" then that should be ok. Thanks for all your guys help with my incessant questions and commits.

**Comment by Night-Pryanik** [comment]
> those pipes || still work as "or"

Yes, they do.

File: src/player_hardcoded_effects.cpp [hunk 2]
```diff
@@ -1001,8 +1001,12 @@ void player::hardcoded_effects( effect &it )
         }
 
         // TODO: Move this to update_needs when NPCs can mutate
+        bool compatible_weather_types = (g->weather == WEATHER_CLEAR || g->weather == WEATHER_SUNNY
```
Comment (Night-Pryanik) [inline src/player_hardcoded_effects.cpp]: I fear gorgon can complain on this (and closing) bracket: there should bean empty space inside brackets, like `( FOO )` They ain't needed, anyway, and can be safely removed.

Comment (patq911) [reply src/player_hardcoded_effects.cpp]: ah, just being cautious.

File: src/player_hardcoded_effects.cpp [hunk 3]
```diff
@@ -1001,8 +1001,12 @@ void player::hardcoded_effects( effect &it )
         }
 
         // TODO: Move this to update_needs when NPCs can mutate
+        bool compatible_weather_types = g->weather == WEATHER_CLEAR || g->weather == WEATHER_SUNNY
```
Comment (ZhilkinSerg) [inline src/player_hardcoded_effects.cpp]: You can move variable declaration to inner scope of next if as it is only used there.

Comment (ZhilkinSerg) [reply src/player_hardcoded_effects.cpp]: Sorry, I've incorrectly read some lines. Everything is fine. Please ignore my comment.

**Pull request merged**